--- a/source/main/gui/RTTLayer.cpp
+++ b/source/main/gui/RTTLayer.cpp
@@ -32,8 +32,8 @@
     MyGUI::xml::ElementEnumerator propert = _node->getElementEnumerator();
     while (propert.next("Property"))
     {
-        const std::string& key = propert->findAttribute("key");
-        const std::string& value = propert->findAttribute("value");
+        std::string key = std::string(propert->findAttribute("key"));
+        std::string value = std::string(propert->findAttribute("value"));
         if (key == "TextureSize")
             setTextureSize(utility::parseValue<IntSize>(value));
         if (key == "TextureName")
