diff --git a/awl/pitchlabel.h b/awl/pitchlabel.h
index 1ef2a3db13..6c4f7bb7f4 100644
--- a/awl/pitchlabel.h
+++ b/awl/pitchlabel.h
@@ -20,6 +20,9 @@
 #ifndef __AWLPITCHLABEL_H__
 #define __AWLPITCHLABEL_H__
 
+#include <QLabel>
+#include <QSize>
+
 namespace Awl {
 
 //---------------------------------------------------------
diff --git a/awl/poslabel.h b/awl/poslabel.h
index 2ea0e6bf65..d904a56897 100644
--- a/awl/poslabel.h
+++ b/awl/poslabel.h
@@ -20,6 +20,9 @@
 #ifndef __AWLPOSLABEL_H__
 #define __AWLPOSLABEL_H__
 
+#include <QLabel>
+#include <QSize>
+
 #include "libmscore/pos.h"
 
 namespace Awl {
@@ -56,4 +59,3 @@ class PosLabel : public QLabel {
 }
 
 #endif
-
diff --git a/libmscore/types.h b/libmscore/types.h
index 2c06b1abf9..3ea77f6f74 100644
--- a/libmscore/types.h
+++ b/libmscore/types.h
@@ -14,6 +14,10 @@
 
 #include "config.h"
 
+#include <QComboBox>
+#include <QObject>
+#include <QString>
+
 /**
  * \namespace Ms .
  */
