diff --git a/mscore/CMakeLists.txt b/mscore/CMakeLists.txt
index 8daa49b517..471cedf06d 100644
--- a/mscore/CMakeLists.txt
+++ b/mscore/CMakeLists.txt
@@ -142,6 +142,7 @@ if (APPLE)
          cocoabridge STATIC
          macos/cocoabridge.mm
          )
+      set_source_files_properties(macos/cocoabridge.mm PROPERTIES SKIP_PRECOMPILE_HEADERS ON)
 else (APPLE)
       set(INCS "")
       set(COCOABRIDGE "")
diff --git a/mscore/macos/cocoabridge.h b/mscore/macos/cocoabridge.h
index d6a217ad99..0c61197bbb 100644
--- a/mscore/macos/cocoabridge.h
+++ b/mscore/macos/cocoabridge.h
@@ -20,6 +20,10 @@
 #ifndef __COCOABRIDGE_H__
 #define __COCOABRIDGE_H__
 
+#include <functional>
+
+#include <QString>
+
 class CocoaBridge {
       CocoaBridge() {};
    public:
