diff --git a/thirdparty/rtf2html/rtf2html.cpp b/thirdparty/rtf2html/rtf2html.cpp
index a494c76ef2..71ea7881fd 100644
--- a/thirdparty/rtf2html/rtf2html.cpp
+++ b/thirdparty/rtf2html/rtf2html.cpp
@@ -47,6 +47,8 @@
 #include <stdexcept>
 #include <fstream>
 #include <string>
+#include <QDebug>
+#include <QString>
 
 QString rtf2html(const QString& iString)
       {
@@ -584,4 +586,3 @@ QString rtf2html(const QString& iString)
             }
       return "";
       }
-
